compiler/ast/src/common/path.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright (C) 2019-2025 Provable Inc.
+// This file is part of the Leo library.
+
+// The Leo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The Leo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the Leo library. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{Expression, Identifier, Node, NodeID, simple_node_impl};
+
+use leo_span::{Span, Symbol};
+
+use itertools::Itertools;
+use serde::{Deserialize, Serialize};
+use std::{fmt, hash::Hash};
+
+/// A Path in a program.
+///
+#[derive(Clone, Default, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub struct Path {
+    /// The path that the user wrote, e.g., `foo::bar`.
+    pub segments: Vec<Identifier>,
+    /// The fully resolved path. We may not know this until the pass PathResolution pass runs.
```

Comment:
```suggestion
    /// The fully resolved path. We may not know this until the `PathResolution` pass runs. 
    /// It is guaranteed to exist after `PathResolution`, after which `segments` should not be used.
```
Is this invariant true?